File: internal/model/booking.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type (
	Booking struct {
		Id              uuid.UUID `json:"id"`
		BookingPeriodId uuid.UUID `json:"bookingPeriodId"`
		Date            time.Time `json:"date"`
	}

	UserBookingItem struct {
		BookingId       uuid.UUID `json:"bookingId"`
		BookingPeriodId uuid.UUID `json:"bookingPeriodId"`
		UserDisplayName string    `json:"userDisplayName"`
		BookingDate     string    `json:"bookingDate"`
	}

	GrouppedUserBooking struct {
		UserDisplayName string            `json:"userDisplayName"`
		Bookings        []UserBookingItem `json:"bookings"`
	}
)

File: web/src/models/booking.ts
export interface BookingPeriod {
    id: number
    from: Date
    to: Date
    isCurrentPeriod: boolean
}

export interface Booking {
    id: number
    userId: string
    date: Date
}

export interface UserBookingItem {
    bookingId: number
    bookingPeriodId: number
    userDisplayName: string
    bookingDate: string
}

export interface UserBooking {
    userDisplayName: string
    bookings: UserBookingItem[]
}